### PR TITLE
Yet more TSDoc

### PR DESCRIPTION
**Pull request by jonathan-codaio**
This is basically all of the rest of the reachable entities. After this, I can send a followup to rework tsdoc_progress.ts as a unittest that enforces that all reachable entities have documentation.

This isn't quite the end of it though, all of the entities that typedoc is currently throwing warnings about will need to be exported and documented too. I'll work on those next.

PTAL @ekoleda-codaio @coda/packs

### CODE REVIEW

File: schema.ts
Comment (ekoleda-codaio) [inline schema.ts:578]: Does this just add build-time checks? What happens though at runtime if you don't return it?

File: schema.ts
Comment (ekoleda-codaio) [inline schema.ts:600]: Some of these links aren't working correctly:

https://coda-packs-sdk--1538.com.readthedocs.build/en/1538/reference/sdk/interfaces/IdentityDefinition/

Should they have been prefixed with a type name?

File: schema.ts
Comment (ekoleda-codaio) [inline schema.ts:619]: Are you missing an "if" here? ", if this is a ..."

File: types.ts
Comment (ekoleda-codaio) [inline types.ts:222]: In my experience in the UI it's labeled with the user's username, not email address.

**Review by ekoleda-codaio — CHANGES_REQUESTED** [review]

**Comment by jonathan-codaio** [comment]
Mind if I merge this and make followups? I'm working on the final cleanup and trying to avoid major rebase conflicts before I get too far.

File: schema.ts
Comment (ekoleda-codaio) [reply schema.ts:578]: The key question I'm getting at is "what happens if I screw up and it's empty?" It sounds like the references that rely on the field won't work, but in general your Pack won't blow up.

File: schema.ts
Comment (ekoleda-codaio) [reply schema.ts]: Just checked the same preview link and those aren't working either.

**Comment by ekoleda-codaio** [comment]
> Mind if I merge this and make followups? I'm working on the final cleanup and trying to avoid major rebase conflicts before I get too far.

Looks like the links are still broken, but once those are fixed feel free to merge.

**Comment by jonathan-codaio** [comment]
> > Mind if I merge this and make followups? I'm working on the final cleanup and trying to avoid major rebase conflicts before I get too far.
> 
> Looks like the links are still broken, but once those are fixed feel free to merge.

Cool, will look into these, but mind approving so I can try to merge if you've already signed off for the night?

**Comment by jonathan-codaio** [comment]
> > > Mind if I merge this and make followups? I'm working on the final cleanup and trying to avoid major rebase conflicts before I get too far.
> > 
> > 
> > Looks like the links are still broken, but once those are fixed feel free to merge.
> 
> Cool, will look into these, but mind approving so I can try to merge if you've already signed off for the night?

Ok these are fixed. Required adding more exports, which in turn required adding more tsdoc, which was pulling on the thread that I was trying to address in my follow-on PR to start dealing with typedoc warnings by adding new exports and entrypoints. Each new import brings in a bunch of new identifiers with it so fixing any single one of these can be deep rabbithole, now that we have doc comment enforcement turned on.

**Review by patrick-codaio — APPROVED** [review]

**Pull request merged**
